File: server/echoes-of-the-metro/src/main/java/com/kevindonovan/eotm/echoes_of_the_metro/models/DTOs/JournalResponse.java
package com.kevindonovan.eotm.echoes_of_the_metro.models.DTOs;

import com.kevindonovan.eotm.echoes_of_the_metro.models.CreatedStatus;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class JournalResponse {

    private int journalId;
    private String title;
    private String text;
    private int storylineId;
    private int appUserId;
    private int whispers;
    private int locationId;
    private Timestamp createdAt;
    private CreatedStatus createdStatus;
}

File: client/eotm-client/src/types/response/JournalResponse.ts
import type { CreatedStatus } from "../enums/CreatedStatus";

export type JournalResponse = {
    journalId: number;
    title: string;
    text: string;
    storylineId: number;
    appUserId: number;
    username: string;
    whispers: number;
    locationId: number;
    createdAt: string;
    createdStatus: CreatedStatus;
}